the-collab-lab/tcl-62-smart-shopping-list
# Jc hs filter list

## Description

This code allows the user to search through their shopping list items to help them find what they're looking for. As the user enters text into the text field, the shopping list data is filtered to only display matches. If the user has entered data that they want to quickly delete, they can tap on the 'x' button. 

## Acceptance criteria
	- A text field (with semantic `<label>`!) which narrows down the list as the user types
	- When there’s text in the field, some kind of button (e.g., an X) to clear the field. When the field is cleared, the list is reset to its unfiltered state.

## Related Issue

closes #6 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria

1. Open the main branch in your terminal and pull all remote changes with `git pull` 
2. Navigate to this feature branch with command `git switch jc-hs-filter-list`  
3. Run `npm start` to spin up the feature in the dev web environment.
4. In your browser, navigate to the List page of the dev website.
5. If the List page is populated with List Items, then continue to step 6. If the List page redirects you to Home, then you will need to set access the development list. 
- Open the devtools in your browser, and navigate to the Local Storage field under the Application tab. You will see empty columns labled 'Key' and 'Value'
- In the application, click 'Create a new list'. This will populate a token for the list. Right click the generated three-word token in the 'Value' column and click 'Edit "Value"'.
- Change the value to "my test list" without quotations. 
- Refresh the page to see the list items populated with data from the development list. 
6.  Begin searching for a shopping list item by typing into the open text field. Below the text field, you will see only list items that include your entered text. Once there are no matches found, you will see that message displayed instead.    
7. Try clicking the 'x' button to clear the text field.
